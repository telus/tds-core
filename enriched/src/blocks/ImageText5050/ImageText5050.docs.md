# ImageText5050 Block
Full-width (in container) content block with image and content that are both 50% width.

## Default (Left-aligned image)
Example:
```javascript
import ImageText5050 from './ImageText5050';

// Example of data structure that needs to be supplied to Component
const data = {
  id: 'google-home-control', // optional, text string
  headline: 'Control smart devices', // optional, text string
  description: 'Link compatible smart home products to control things like your thermostat or entertainment on your TV.*', // required, text string
  legalText: '*Controlling your TV requires a Chromecast or a TV with Chromecast built-in.', // optional, text string
  image: { // Standard media field from Contentful
    file: {
      url: 'https://images.contentful.com/oia8yg65m0on/46kiEmr0pa2EsuSIWIA8YS/1b1626894d0957a22b00b6afe0428ead/nest-google-home__1_.jpg'
    },
    description: 'Photo showing Nest thermostat beside Google Home'
  },
  imageAlign: 'Left' // supports "Left" or "Right". Defaults Left. text string
};

<ImageText5050 content={data} />
```

```
<ImageText5050 content={{
  id: 'google-home-control',
  headline: 'Control smart devices',
  description: 'Link compatible smart home products to control things like your thermostat or entertainment on your TV.*',
  legalText: '*Controlling your TV requires a Chromecast or a TV with Chromecast built-in.',
  image: {
    file: {
      url: 'https://images.contentful.com/oia8yg65m0on/46kiEmr0pa2EsuSIWIA8YS/1b1626894d0957a22b00b6afe0428ead/nest-google-home__1_.jpg'
    },
    description: 'Photo showing Nest thermostat beside Google Home'
  },
  imageAlign: 'Right'
}} />
```

## Right-aligned image
Example:
```javascript
import ImageText5050 from './ImageText5050';

// Exact same as above with an additional image align flag
const data = {
  ...
  imageAlign: 'Right'
};

<ImageText5050 content={data} />
```

```
<ImageText5050 content={{
  id: 'google-home-control',
  headline: 'Control smart devices',
  description: 'Link compatible smart home products to control things like your thermostat or entertainment on your TV.*',
  legalText: '*Controlling your TV requires a Chromecast or a TV with Chromecast built-in.',
  image: {
    file: {
      url: 'https://images.contentful.com/oia8yg65m0on/46kiEmr0pa2EsuSIWIA8YS/1b1626894d0957a22b00b6afe0428ead/nest-google-home__1_.jpg'
    },
    description: 'Photo showing Nest thermostat beside Google Home'
  },
  imageAlign: 'Right'
}} />
```
