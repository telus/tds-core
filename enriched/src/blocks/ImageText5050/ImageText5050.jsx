// Dependencies
import React from 'react';
import PropTypes from 'prop-types';

// Assets
import './ImageText5050.scss';

const ImageText5050 = ({ content }) => {
  return (
    <div
      id={content.id ? content.id : ''}
      className={`image_text_50-50 ${content.imageAlign === 'Right' ? 'align-right' : ''} container container--fluid`}>
      <div className="image_text_50-50__content image_text_50-50__image">
        <img
          src={content.image.file.url}
          alt={content.image.description} />
      </div>
      <div className="image_text_50-50__content image_text_50-50__copy">
        <div className="inner-copy">
          {
            content.headline &&
            <h2 tabIndex="0" className="image_text_50-50__headline">{content.headline}</h2>
          }
          <p className="text--medium">{content.description}</p>
          {
            content.legalText &&
            <p className="text--small">{content.legalText}</p>
          }
        </div>
      </div>
    </div>
  );
};

ImageText5050.propTypes = {
  content: PropTypes.object.isRequired
};


export default ImageText5050;
