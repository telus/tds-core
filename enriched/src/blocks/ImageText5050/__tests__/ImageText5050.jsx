import React from 'react';
import { shallow, render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import ImageText5050 from '../ImageText5050';


describe('ImageText5050', () => {
  const content = {
    id: 'id',
    description: 'description',
    headline: 'headline',
    legalText: 'legal text',
    image: {
      file: {
        url: 'file url'
      },
      description: 'file description'
    }
  };

  const imageText5050 = shallow(<ImageText5050 content={content} />);

  it('should render ImageText5050 (snapshot)', () => {
    expect(renderToJson(imageText5050)).toMatchSnapshot();
  });

  it('should render headline if it exists', () => {
    expect(imageText5050.find('.image_text_50-50__headline').length).toEqual(1);
  });

  it('should render legal text if it exists', () => {
    expect(imageText5050.find('.image_text_50-50__copy').html()).toContain(content.legalText);
  });

  it('should add flag to align content right if flag is set to Right', () => {
    content.imageAlign = "Right";

    const newImageText5050 = shallow(<ImageText5050 content={content} />);

    expect(newImageText5050.find('.align-right').length).toEqual(1);
  });
});
