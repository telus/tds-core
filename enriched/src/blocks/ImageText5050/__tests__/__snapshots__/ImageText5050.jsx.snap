// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageText5050 should render ImageText5050 (snapshot) 1`] = `null`;
